src/dataframe.rs

```diff
@@ -36,7 +36,7 @@ use std::sync::Arc;
 /// The actual execution of a plan runs natively on Rust and Arrow on a multi-threaded environment.
 #[pyclass(name = "DataFrame", module = "datafusion", subclass)]
 #[derive(Clone)]
-pub(crate) struct PyDataFrame {
+pub struct PyDataFrame {
```

Comment:
This is also due to 34.0.0?

Reply:
I don't think directly. I will say having lots of these "(crate)" level visibility of structs makes using ADP as a library difficult since they are not visible outside the crate. I had started removing them awhile back and think Andy was just removing another one of them. I should probably just make a PR to remove them all, where it makes sense of course

Reply:
I removed the `(crate)` restriction to work around many warnings like the following. I wonder if this was caused by a recent Rust version upgrade rather than anything related to 34.0.0.

```
warning: type `PyDataFrame` is more private than the item `PySessionContext::sql`
   --> src/context.rs:282:5
    |
282 |     pub fn sql(&mut self, query: &str, py: Python) -> PyResult<PyDataFrame> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method `PySessionContext::sql` is reachable at visibility `pub`
    |
note: but type `PyDataFrame` is only usable at visibility `pub(crate)`
   --> src/dataframe.rs:39:1
    |
39  | pub(crate) struct PyDataFrame {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: `#[warn(private_interfaces)]` on by default
```

Reply:
There is a reference to `private_interfaces` in the [release notes](https://releases.rs/docs/1.74.0/) for Rust 1.74.0, which was released last month.

Reply:
I see. Thanks for the reference.